car.js

```diff
@@ -135,10 +135,38 @@ async function writeStream (stream) {
   return new CarDatastore(reader, writer)
 }
 
+async function traverseBlock (block, get, car, concurrency = 1, seen = new Set()) {
+  const cid = await block.cid()
+  await car.put(cid, block.encodeUnsafe())
+  seen.add(cid.toString('base58btc'))
+  if (cid.codec === 'raw') return
+  const reader = block.reader()
+  const missing = link => !seen.has(link.toString('base58btc'))
```

Comment:
One minor style nit that grates on me is arrow functions without parens. `(link) => ...` to make it crystal clear what's going on here